engine/src/scheme.rs

```diff
@@ -1,29 +1,53 @@
 use crate::{
     ast::FilterAst,
     functions::Function,
-    lex::{complete, expect, span, take_while, LexErrorKind, LexResult, LexWith},
-    types::{GetType, Type},
+    lex::{complete, expect, skip_space, span, take_while, LexErrorKind, LexResult, LexWith},
+    types::{GetType, RhsValue, Type},
 };
 use failure::Fail;
 use fnv::FnvBuildHasher;
 use indexmap::map::{Entry, IndexMap};
-use serde::{Deserialize, Serialize, Serializer};
+use serde::{ser::SerializeSeq, Deserialize, Serialize, Serializer};
 use std::{
     cmp::{max, min},
     error::Error,
     fmt::{self, Debug, Display, Formatter},
+    iter::Iterator,
     ptr,
 };
 
-#[derive(PartialEq, Eq, Clone, Copy)]
+#[derive(Debug, PartialEq, Eq, Clone, Serialize)]
+#[serde(un$tagged)]
+pub enum FieldPathItem {
```

Comment:
I was thinking if we can find a name for this Enum that is more explicit about the fact it's purpose is to access elements inside LhsValue.

Reply:
Hmm. Why does this use an enum with a single variant in the first place?

Reply:
We will add a second variant within the next week or so to support array indexing.